R/Private.R

```diff
@@ -0,0 +1,88 @@
+.getPreferenceScoreDist <- function(connectionDetails = NULL,
+                                    preferenceScoreDist = NULL,
+                                    targetId,
+                                    comparatorId,
+                                    databaseId,
+                                    analysisId) {
+  if (!is.null(connectionDetails)) {
+    if (exists("connectionDetails@schema")) {
+      schema <- connectionDetails@schema
+    } else {
+      schema <- NULL
+    }
+    connection <- DatabaseConnector::connect(connectionDetails)
+    sql <-   "SELECT *
+              FROM  {@schema !=} ? {@schema.preference_score_dist}:{preference_score_dist}
+              WHERE target_id = @targetId
+            	AND comparator_id = @comparatorId
+            	AND database_id = '@databaseId'
+            	AND analysis_id = @analysisId;"
+    
+    sql <-
+      SqlRender::render(
+        sql = sql,
+        targetId = targetId,
+        comparatorId = comparatorId,
+        schema = schema,
+        databaseId = databaseId,
+        analysisId = analysisId
+      )
+    sql <-
+      SqlRender::translate(sql = sql, targetDialect = connection@dbms)
+    preferenceScoreDist <-
+      DatabaseConnector::querySql(
+        connection = connection,
+        sql = sql,
+        snakeCaseToCamelCase = TRUE
+      )
+  } else {
+    preferenceScoreDist <- preferenceScoreDist %>%
+      dplyr::filter(
+        targetId %in% !!targetId,
+        comparatorId %in% !!comparatorId,
+        databaseId %in% !!databaseId,
+        analysisId %in% !!analysisId
+      ) %>%
+      dplyr::select(preferenceScore, targetDensity, comparatorDensity)
+  }
+  return(preferenceScoreDist)
+}
+
+
+.getExposureOfInterest <- function(connectionDetails = NULL,
+                                   exposureOfInterest = NULL,
+                                   targetId ,
+                                   comparatorId) {
+  if (!is.null(connectionDetails)) {
+    if (exists("connectionDetails@schema")) {
+      schema <- connectionDetails@schema
+    } else {
+      schema <- NULL
+    }
+    connection <- DatabaseConnector::connect(connectionDetails)
+    sql <-   "SELECT exposure_id, exposure_name
+              FROM {@schema !=} ? {exposure_of_interest}:{exposure_of_interest}
```

Comment:
This if-then-else statement doesn't do anything (both options are identical). Note that the [recommendation](https://ohdsi.github.io/SqlRender/articles/UsingSqlRender.html#schemas-and-databases) is to always specify the database schema explicitly. So in this case I would write this SQL as:

```sql
SELECT exposure_id, exposure_name
FROM @resultDatabaseSchema.exposure_of_interest
WHERE exposure_id IN (@targetId, @comparatorId);

```